rgw/admin/bucket.go

```diff
@@ -0,0 +1,153 @@
+			SizeUtilized   *uint64 `json:"size_utilized"`
+			SizeKb         *uint64 `json:"size_kb"`
+			SizeKbActual   *uint64 `json:"size_kb_actual"`
+			SizeKbUtilized *uint64 `json:"size_kb_utilized"`
+			NumObjects     *uint64 `json:"num_objects"`
+		} `json:"rgw.multimeta"`
+	} `json:"usage"`
+	BucketQuota struct {
+		Enabled    *bool   `json:"enabled"`
+		CheckOnRaw *bool   `json:"check_on_raw"`
+		MaxSize    *uint64 `json:"max_size"`
+		MaxSizeKb  *uint64 `json:"max_size_kb"`
+		MaxObjects *uint64 `json:"max_objects"`
+	} `json:"bucket_quota"`
+	Policy bool `url:"policy"`
+}
+
+// Policy describes a bucket policy
+type Policy struct {
+	ACL struct {
+		ACLUserMap []struct {
+			User string `json:"user"`
+			ACL  *int   `json:"acl"`
+		} `json:"acl_user_map"`
+		ACLGroupMap []interface{} `json:"acl_group_map"`
+		GrantMap    []struct {
+			ID    string `json:"id"`
+			Grant struct {
+				Type struct {
+					Type int `json:"type"`
+				} `json:"type"`
+				ID         string `json:"id"`
+				Email      string `json:"email"`
+				Permission struct {
+					Flags int `json:"flags"`
+				} `json:"permission"`
+				Name    string `json:"name"`
+				Group   *int   `json:"group"`
+				URLSpec string `json:"url_spec"`
+			} `json:"grant"`
+		} `json:"grant_map"`
+	} `json:"acl"`
+	Owner struct {
+		ID          string `json:"id"`
+		DisplayName string `json:"display_name"`
+	} `json:"owner"`
+}
+
+// ListBuckets will return the list of all buckets present in the object store
+func (api *API) ListBuckets(ctx context.Context) ([]string, error) {
+	body, err := api.call(ctx, get, "/bucket", nil)
+	if err != nil {
+		return nil, err
+	}
+	var s []string
+	err = json.Unmarshal(body, &s)
+	if err != nil {
+		return nil, fmt.Errorf("%s. %s. %w", unmarshalWrapError, string(body), err)
+	}
+
+	return s, nil
+}
+
+// GetBucketInfo will return various information about a specific token
+func (api *API) GetBucketInfo(ctx context.Context, bucket string) (*Bucket, error) {
+	b := Bucket{
+		Bucket: bucket,
+	}
+	body, err := api.call(ctx, get, "/bucket", valueToURLParams(b))
+	if err != nil {
+		return nil, err
+	}
+
+	ref := &Bucket{}
+	err = json.Unmarshal(body, ref)
+	if err != nil {
+		return nil, fmt.Errorf("%s. %s. %w", unmarshalWrapError, string(body), err)
+	}
+
+	return ref, nil
+}
+
+// GetBucketPolicy - http://docs.ceph.com/docs/mimic/radosgw/adminops/#get-bucket-or-object-policy
+func (api *API) GetBucketPolicy(ctx context.Context, bucket string) (*Policy, error) {
+	b := Bucket{
+		Bucket: bucket,
+		Policy: true,
+	}
+	body, err := api.call(ctx, get, "/bucket", valueToURLParams(b))
+	if err != nil {
+		return nil, err
+	}
+
+	ref := &Policy{}
+	err = json.Unmarshal(body, ref)
+	if err != nil {
+		return nil, fmt.Errorf("%s. %s. %w", unmarshalWrapError, string(body), err)
+	}
+
+	return ref, nil
+}
+
+// RemoveBucket will remove a given token from the object store
+func (api *API) RemoveBucket(ctx context.Context, bucket string) error {
```

Comment:
@leseb : If you are doing another round of refreshing PR, please include `purge-objects` option for this api